src/app/market/buy/checkout-process/checkout-process.component.html

```diff
@@ -138,6 +138,10 @@
 
   <!-- =========== SHIPPING DETAILS =========== -->
   <mat-step [completed]="cache.selectedIndex >=2" [stepControl]="shippingFormGroup">
+    <button mat-raised-button color="primary" [disabled]="!profile?.shippingAddresses || profile?.shippingAddresses.length === 0" (click)="clearForm()">
```

Comment:
I think the button should probably be disabled if no saved profile shipping profile has been selected. In other words, If a saved profile has not been selected, then this button should be disabled. 

Alternatively, change the button text to read something similar to 'Clear Shipping Address form', as that is essentially its purpose. In which case it would make sense to disable the button if the entire form is empty, and it would be better then to move the button to the form that it clears.